### PR TITLE
feature(ui-ux): update address book UI and enhance features

### LINKED ISSUE #2654: `address book` - updated designs
#### What would you like to be added:
Update design of address book and some additional features. 

#### Send Screen
- [ ] Address input with explicit "Paste" button
image

- [ ] If an address is from address book (whether pasted, from address book or QR scan - display the label). The purpose of this feature is to ensure users that whatever they're inputting is what they have on the address book. Serves as a cross check for them.
image

- [ ] On delete of an Address - Show toast  - "Address deleted"

- [ ] When a user adds an address, it should close the modal and "pre-selects" this address immediately. This behavior is only on Send Screen and not on settings screen.

#### Settings Screen

- [ ] Settings -> Address Book - Add CTA 
image

#### Address Book Screen

- [ ] Updated input of address
image

- [ ] Updated copy and display of Address book screen
image

- [ ] Empty State
image

- [ ] Search from the header

### PR DESCRIPTION
**Pull request by kyleleow**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:
- To update address book UI
- To automatically select new address upon successfully added in address book
- To add address book entry point in setting screen
- To display matched label in address input
- To display matched address indicator below address input
- To allow search in address book

#### Which issue(s) does this PR fixes?:

Fixes #2654

#### Developer Checklist:

- [x] Read your code changes at least once
- [x] Tested on iOS/Android device (e.g, No crashes, library supported etc.)
- [x] No console errors on web
- [x] Tested on Light mode and Dark mode*
- [x] Your UI implementation visually matched the rendered design*
- [ ] Unit tests*
- [ ] Added e2e tests*
- [ ] Added translations*

### CODE REVIEW

File: mobile-app/app/components/WalletTextInput.tsx [hunk 1]
```diff
@@ -66,13 +75,32 @@ export const WalletTextInput = forwardRef<any, WalletTextInputProps>(function (p
...
+          <ThemedSectionTitle
+            light={tailwind('text-gray-700')}
+            dark={tailwind('text-gray-200')}
+            testID={titleTestID}
+            text={title}
+            style={tailwind('text-base')}
+          />
+          {pasteButton?.onPasteButtonPress !== undefined && (
+            <TouchableOpacity
+              // eslint-disable-next-line
```
Comment (thedoublejay) [inline mobile-app/app/components/WalletTextInput.tsx:89]: why?

File: mobile-app/app/screens/AppNavigator/screens/Balances/screens/AddressBookScreen.tsx [hunk 2]
```diff
@@ -21,6 +20,12 @@ import { useWalletNodeContext } from '@shared-contexts/WalletNodeProvider'
 import { useLogger } from '@shared-contexts/NativeLoggingProvider'
 import { MnemonicStorage } from '@api/wallet/mnemonic_storage'
 import { authentication, Authentication } from '@store/authentication'
+import { Button } from '@components/Button'
+import { HeaderSearchIcon } from '@components/HeaderSearchIcon'
+import { HeaderSearchInput } from '@components/HeaderSearchInput'
+import { openURL } from 'expo-linking'
```
Comment (thedoublejay) [inline mobile-app/app/screens/AppNavigator/screens/Balances/screens/AddressBookScreen.tsx:26]: can we use openURL from linking.ts?

File: mobile-app/app/screens/AppNavigator/screens/Balances/screens/AddressBookScreen.tsx [hunk 3]
```diff
@@ -33,6 +38,28 @@ export function AddressBookScreen ({ route, navigation }: Props): JSX.Element {
   const userPreferences = useSelector((state: RootState) => state.userPreferences)
   const addressBook = userPreferences.addressBook
   const [isEditing, setIsEditing] = useState(false)
+  const { getAddressUrl } = useDeFiScanContext()
+
+  // Search
+  const [showSearchInput, setShowSearchInput] = useState(false)
+  const [searchString, setSearchString] = useState('')
+  const filterAddress = debounce((searchString: string): void => {
+    if (searchString !== undefined && searchString.trim().length > 0) {
```
Comment (thedoublejay) [inline mobile-app/app/screens/AppNavigator/screens/Balances/screens/AddressBookScreen.tsx:47]:
```suggestion
    if (searchString?.trim().length > 0) {
```

File: mobile-app/app/screens/AppNavigator/screens/Balances/screens/AddressBookScreen.tsx [hunk 4]
```diff
@@ -33,6 +38,28 @@ export function AddressBookScreen ({ route, navigation }: Props): JSX.Element {
...
+      const addressBookList: string[] = []
+
+      for (const address in addressBook) {
+        if (address.includes(searchString.trim().toLowerCase()) || addressBook[address].label.toLowerCase().includes(searchString.trim().toLowerCase())) {
```
Comment (thedoublejay) [inline mobile-app/app/screens/AppNavigator/screens/Balances/screens/AddressBookScreen.tsx:51]: Might want to check if this exist = addressBook[address]

**Review by thedoublejay — CHANGES_REQUESTED** [review]

File: mobile-app/app/components/WalletTextInput.tsx (hunk 1 above)
Comment (kyleleow) [reply mobile-app/app/components/WalletTextInput.tsx:89]: to supress the handle naming error `Handler function for onPress prop key must be a camelCase name beginning with 'handle' only `

File: mobile-app/app/screens/AppNavigator/screens/Balances/screens/AddressBookScreen.tsx (hunk 2 above)
Comment (kyleleow) [reply mobile-app/app/screens/AppNavigator/screens/Balances/screens/AddressBookScreen.tsx]: updated

File: mobile-app/app/screens/AppNavigator/screens/Balances/screens/AddressBookScreen.tsx (hunk 3 above)
Comment (kyleleow) [reply mobile-app/app/screens/AppNavigator/screens/Balances/screens/AddressBookScreen.tsx]: updated

File: mobile-app/app/screens/AppNavigator/screens/Balances/screens/AddressBookScreen.tsx (hunk 4 above)
Comment (kyleleow) [reply mobile-app/app/screens/AppNavigator/screens/Balances/screens/AddressBookScreen.tsx]: added optional chaining when accessing nested label

**Pull request merged**